### PR TITLE
Update Dockerfile and add RBAC rules needed to run operator in cluster

### PR DESCRIPTION
**Pull request by akrejcir**
**What this PR does / why we need it**:
```release-note
NONE
```

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -18,11 +18,10 @@ COPY internal/ internal/
 # Build
 RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on go build -a -o manager main.go
 
-# Use distroless as minimal base image to package the manager binary
-# Refer to https://github.com/GoogleContainerTools/distroless for more details
-FROM gcr.io/distroless/static:nonroot
+FROM registry.access.redhat.com/ubi8/ubi-minimal
```
Comment (omeryahud) [inline Dockerfile:21]: Why the change of the base image?

File: config/manager/manager.yaml [hunk 2]
```diff
@@ -32,8 +32,8 @@ spec:
         resources:
           limits:
             cpu: 100m
-            memory: 30Mi
+            memory: 100Mi
```
Comment (omeryahud) [inline config/manager/manager.yaml:35]: Please don't specify resource limits and requests before we have a proven performance issue (please remove them completely, even the default ones).

We did it before with `template-validator`, and it just made our life difficult when deploying on smaller clusters and debugging.

**Review by omeryahud — CHANGES_REQUESTED** [review]
Two small things, other than that lgtm

File: Dockerfile (hunk 1 above)
Comment (akrejcir) [reply Dockerfile:21]: Other kubevirt projects use `ubi8` as a base for containers.

Comment (omeryahud) [reply Dockerfile:21]: Ah, very well, nice catch

File: config/manager/manager.yaml (hunk 2 above)
Comment (akrejcir) [reply config/manager/manager.yaml:35]: Ok, I will remove them. I hit the OOM error with the default limits. On my machine the controller need about 56 Mi.

Comment (akrejcir) [reply config/manager/manager.yaml]: Done

**Review by omeryahud — APPROVED** [review]
/lgtm
